### DISCUSSION TITLE
whereMorphedTo('sectionable', null)

### DISCUSSION THREAD

**Discussion by m-derakhshi** [laravel/framework#46812] [Q&A]
hi

my query is : 
`
select * from visitor_info` where `selectable_type` is null
`

i want use eloquent and phpstorm autocomplate:
`
Model:query()->whereMorphedTo('sectionable', null)->get();
`

laravel error:
`
Call to a member function getMorphClass() on null
`

**Comment by Angel5a** [comment]
Just use:
`Model:query()->whereDoesntHaveMorph('sectionable', '*')->get();`

**Reply by Orrison** [reply]
This does not work. `whereDoesntHaveMorph` will always add a `"model"."morph_type" = ? and not exists (select *...`

So `whereDoesntHaveMorph` cannot be used to check for non-existence of records in a `MorphTo`. Only checks in existing records.
